### ISSUE TITLE
sfs_load fails to find some sfs

### ISSUE THREAD

**Issue by mavrothal** [puppylinux-woof-CE/woof-CE#1052]
[line 721](https://github.com/puppylinux-woof-CE/woof-CE/blob/testing/woof-code/rootfs-skeleton/usr/sbin/sfs_load#L721) will eliminate any SFS with a name "*-NNN.sfs" (where N=[0-9]).
Notably the slacko kernel source SFSs that ends with "-700.sfs"
I'm not sure that this is the best way to eliminate other compat-distro SFSs or if we can even find a user-proof way to avoid loading an SFS that will mess-up the running system but you may want to consider additional tests (ie presence of both "puppy|*drv" and "*-NNN.sfs") or just commenting out this line. 
Or maybe change kernel source file names!...

**Comment by gyrog** [comment]
I would suggest, comment it out.
If the default sfs file names, defined in DISTRO_SPECS ,and the actual files used, as defined in PUPSTATE, are ignored, that should be enough.

**Comment by gyrog** [comment]
Hmm..., the more I think about it, sfs_load really only neds to avoid the sfs's that are aleady loaded by "init". i.e. the ones defined in PUPSTATE, and even then it's only to avoid wasting effort.

**Comment by wdlkmpx** [comment]
the logic to ignore sfs_files is very simple. sfs_load ignores both the files specified in distro_specs and pupstate and many more. i'll remove a few lines.

**Comment by wdlkmpx** [comment]
it should be fixed now. there should be a puplib script or something for scripts like sfs_load to source it. sfs_load duplicates code already present in /usr/local/pup_event, it also has many functions that can be used in many other scripts or that are already implemented in other scripts one way or another.

its design to be an app for all puppy versions and derivative works does not apply anymore.

**Comment by mavrothal** [comment]
However, the risk of messing up a system by loading another puppy main sfs is real, given the hundred of puppies around. As I said is hard to user-proof system but maybe we can at least test if an SFS to be mounted has an /etc/DISTO_SPECS file and refuse get it into the union if so.

**Comment by gyrog** [comment]
Why should I not be allowed to load the main sfs of another puppy as an extra sfs?
Given the sfs_load user interface, it's highly unlikely, unless I've managed to store it in the same directory as my current puppy.
And even if I did, what disaster will befall me, well most likely nothing.
It gets loaded at the bottom of the stack, most of it's files would be covered by the files in the real puppy...sfs, and if they weren't then it's unlikely that the programs in the current puppy would access them.
And even if it did produce a disaster, it's my own stupid fault.
